runtime/src/accounts_db.rs

```diff
@@ -1032,14 +1051,20 @@ impl AccountsDB {
                         if forced { " (forced)" } else { "" },
                     );
                     return 0;
-                } else if (alive_count as f32 / stored_count as f32) >= 0.80 && !forced {
-                    trace!(
-                        "shrink_stale_slot ({}): not enough space to shrink: {} / {}",
-                        slot,
-                        alive_count,
-                        stored_count,
-                    );
-                    return 0;
+                } else if !forced {
+                    let sparse_by_count = (alive_count as f32 / stored_count as f32) <= 0.8;
+                    let sparse_by_space = (written_space as f32 / total_space as f32) <= 0.8;
+                    if !sparse_by_count && !sparse_by_space {
+                        trace!(
```

Comment:
hmm, should I change to `info!`; not sure.. time to bloat our INFO logs further...? ;)

Reply:
yeah, I took the bitter path. ;)

Reply:
yeah, I took the bitter path. ;)